great_expectations/checkpoint/actions.py

```diff
@@ -81,6 +82,30 @@ def _build_renderer(config: dict) -> Renderer:
     return renderer
 
 
+class ActionContext:
+    """
+    Shared context for all actions in a checkpoint run.
+    Note that order matters in the action list, as the context is updated with each action's result.
+    """
+
+    def __init__(self) -> None:
+        self._data: list[tuple[ValidationAction, dict]] = []
+
+    def update(self, action: ValidationAction, action_result: dict) -> None:
+        self._data.append((action, action_result))
+
+    def filter_results(self, class_: Type[ValidationAction]) -> list[dict]:
+        return [action_result for action, action_result in self._data if action.__class__ is class_]
```

Comment:
nit: why not just us isinstance?

Reply:
I didn't have coffee yet is why 😆 I'll change it